warehouse/warehouse.go

```diff
@@ -1447,12 +1405,23 @@ func Start(ctx context.Context, app app.App) error {
 		}
 	}()
 
+	g, ctx := errgroup.WithContext(ctx)
```

Comment:
@achettyiitr it feels like we don't need a group here since none of the two routines are returning an error. also, we're overriding the context with the group one which could cause confusion in the code after the group that might be using the context. wdyt?